### ISSUE TITLE
Snippet functionality not available in new file pre save

### ISSUE THREAD

**Issue by kro12** [zed-industries/zed#13636]
### Check for existing issues

- [X] Completed

### Describe the feature

Wasn't sure if this should go down as a bug or a feature request, but I've given the benefit of the doubt here.

I'm a user who has recently come over from VS Code, and I had a few snippets which I was glad to realise I could start using since [this](https://github.com/zed-industries/zed/issues/4611) dropped.

I have a very basic snippet that generates a new React Component boilerplate for me, just to save a little typing as I will often import from our internal libraries, etc. I tried this out in Zed today and at first could not figure out why it wasn't working - after installing the extension and ensuring my snippets were saved to the correct location.

The steps I took were:

Created a new file
Typed the snippet shortcut
and nothing would happen. Then I thought perhaps I need to set a file type but alas, that didn't help either. It then occurred to me to try it on an existing file and lo and behold, it worked there.

So, obviously this isn't a deal-breaker, I can create the new file, save it immediately, and then the snippet is accessible to me. It's more of a minor quality of life improvement if it's a change that's easily rolled out.

**Comment by osiewicz** [comment]
This is a general limitation of our Language Server support; we do not query language servers in unsaved files. It's not a snippets extension limitation per se.

**Comment by kro12** [comment]
Ah okay, I get you. Thanks for the quick follow up comment.

**Comment by osiewicz** [comment]
We're most likely gonna get rid of snippets-as-extension though and bake it into the core editor instead; in that case, we may be able to get around this limitation of ours. :)

**Comment by kro12** [comment]
Thanks for the additional info @osiewicz - judging by the phrasing I suppose there's no real timeline on these potential changes?

**Comment by osiewicz** [comment]
No, quite the contrary; https://github.com/zed-industries/zed/pull/13937
It's a matter of hours before this gets merged, and then, depending on whether you use Preview or Stable builds, you'll either be able to use snippets in unsaved files starting tomorrow or in a week. :)

**Comment by kro12** [comment]
Excellent, thanks @osiewicz!

**Comment by fuzenco** [comment]
I can confirm snippets work now without saving the file first. Thanks team!
